src/engine/N3Base/N3UIEdit.cpp

```diff
@@ -752,10 +758,22 @@ void CN3UIEdit::UpdateCaretPosFromEditCtrl()
 		ReleaseDC(s_hWndEdit, hDC);
 	}
 */
-	int iTmp = ::SendMessage(s_hWndEdit, EM_GETSEL, 0, 0);
-	int iCaret = LOWORD(iTmp);
-	int iCTmp2 = HIWORD(iTmp);
-	s_pFocusedEdit->SetCaretPos(iCaret);
+	int cursorPosition = 0;
+
+	if (changeFocus == false) {
+		int iTmp = ::SendMessage(s_hWndEdit, EM_GETSEL, NULL, NULL);
+		cursorPosition = LOWORD(iTmp);
+		int iCTmp2 = HIWORD(iTmp);
+	}
+	else {
+		std::string getText = s_pFocusedEdit->GetString();
+		cursorPosition = getText.size();
+
+		SendMessageA(s_hWndEdit, EM_SETSEL, (WPARAM)cursorPosition, (LPARAM)cursorPosition);
+	}
+
+
+	s_pFocusedEdit->SetCaretPos(cursorPosition);
```

Comment:
We could write the same thing like this:
```suggestion
	int cursorPosition = 0;
	if (changeFocus) {
		cursorPosition = s_pFocusedEdit->GetString().size();
		SendMessage(s_hWndEdit, EM_SETSEL, (WPARAM)cursorPosition, (LPARAM)cursorPosition);
	} else {
		cursorPosition = LOWORD(SendMessage(s_hWndEdit, EM_GETSEL, NULL, NULL));
	}

	s_pFocusedEdit->SetCaretPos(cursorPosition);
```

I'm not sure however if `EM_SETSEL` is the right pick here. I would have to check further to tell.

Reply:
I that function there was variable int iLen = GetWindowTextLength(s_hWndEdit); too get editbox length but this variable is never used  and it was commented out